lib/core/src/model.rs

```diff
@@ -735,3 +735,68 @@ macro_rules! get_invoice_amount {
             / 1000
     };
 }
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub(crate) struct SwapUpdateTxDetails {
+    #[serde(rename(deserialize = "id", serialize = "id"))]
+    pub(crate) tx_id: String,
+    pub(crate) hex: String,
+}
+
+/// Intermediate representation of [boltz_client::swaps::boltzv2::Update]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub(crate) enum Update {
+    TransactionMempool {
+        id: String,
+        status: String,
+        transaction: SwapUpdateTxDetails,
+    },
+    Generic {
+        id: String,
+        status: String,
+    },
+}
+
+impl Update {
+    pub(crate) fn get_swap_id(&self) -> &String {
+        match self {
+            Update::Generic { id, .. } => id,
+            Update::TransactionMempool { id, .. } => id,
+        }
+    }
+
+    pub(crate) fn get_swap_state(&self) -> &String {
+        match self {
+            Update::Generic { status, .. } => status,
+            Update::TransactionMempool { status, .. } => status,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct RespError {
+    pub(crate) id: String,
+    pub(crate) error: String,
+}
+
+/// Intermediate representation of [boltz_client::swaps::boltzv2::SwapUpdate]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(untagged)]
+pub(crate) enum SwapUpdate {
+    Subscription {
+        event: String,
+        channel: String,
+        args: Vec<String>,
+    },
+    Update {
+        event: String,
+        channel: String,
+        args: Vec<Update>,
+    },
+    Error {
+        event: String,
+        channel: String,
+        args: Vec<RespError>,
+    },
+}
```

Comment:
It seems that all these models belong to the swapper module.

Reply:
@hydra-yse  actually I see they are copied from boltz client, any reason why not fix there?

Reply:
I asked him to first do it here and get it working.

Then @hydra-yse this might be a good time to add it upstream and include it in our patch branch.